src/rpc/actor_registry.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019-2025 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+use crate::lotus_json::HasLotusJson;
+use crate::networks::ACTOR_BUNDLES_METADATA;
+use crate::shim::actors::{
+    AccountActorStateLoad, CronActorStateLoad, EVMActorStateLoad, MarketActorStateLoad,
+    MinerActorStateLoad, SystemActorStateLoad, account, cron, evm, market, miner, system,
+};
+use crate::shim::machine::BuiltinActor;
+use ahash::{HashMap, HashMapExt};
+use anyhow::anyhow;
+use cid::Cid;
+use fvm_ipld_blockstore::Blockstore;
+use once_cell::sync::Lazy;
+use serde_json::Value;
+
+// Build a map from CIDs to actor types and versions once at startup
+static CID_TO_ACTOR_TYPE: Lazy<HashMap<Cid, (BuiltinActor, u64)>> = Lazy::new(|| {
+    let mut map = HashMap::new();
+
+    // Populate the map using data from ACTOR_BUNDLES_METADATA
+    for ((_, _), metadata) in ACTOR_BUNDLES_METADATA.iter() {
+        if let Ok(version) = metadata.actor_major_version() {
+            for (actor_type, cid) in metadata.manifest.builtin_actors() {
+                map.insert(cid, (actor_type, version));
+            }
+        }
+    }
+
+    map
+});
+
+pub fn get_actor_type_from_code(code_cid: &Cid) -> anyhow::Result<(BuiltinActor, u64)> {
+    CID_TO_ACTOR_TYPE
+        .get(code_cid)
+        .copied()
+        .ok_or_else(|| anyhow!("Unknown actor code CID: {}", code_cid))
+}
```

Comment:
nit: it'd be nice to have it as a separate struct so we can abstract away the underlying data structure. It'd only have `read_actor_type(cid: &Cid)` as a member method.

A unit test would also be nice to have.